src/smt/theory_str_noodler/aut_assignment.cpp

```diff
@@ -158,4 +158,34 @@ namespace smt::noodler {
             cur_level = next_level;
         }
     }
+
+    bool AutAssignment::is_flat(const BasicTerm& t) const {
+        bool flat = true;
+
+        mata::nfa::Nfa aut = *this->at(t);
+
+        mata::nfa::Nfa::TarjanDiscoverCallback callback {};
+        callback.scc_discover = [&](const std::vector<mata::nfa::State>& scc, const std::vector<mata::nfa::State>& tarjan_stack) -> bool {
```

Comment:
It is better to be explicit which variables we are capturing.
```suggestion
        callback.scc_discover = [&flat](const std::vector<mata::nfa::State>& scc, const std::vector<mata::nfa::State>& tarjan_stack) -> bool {
```

Reply:
Can you also add comment what this function does?

Reply:
I added a comment ot `is_flat` method where it should be explained.